src/mlagility/api/cloud.py

```diff
@@ -175,55 +175,84 @@ def setup_groqchip_host(client) -> None:
     with MySFTPClient.from_transport(client.get_transport()) as s:
         s.put(f"{dir_path}/execute.py", "groqflow_remote_cache/execute.py")
 
-
-def setup_gpu_host(client) -> None:
-    # Check if at least one NVIDIA GPU is available remotely
-    stdout, exit_code = exec_command(client, "lspci | grep -i nvidia")
-    if stdout == "" or exit_code == 1:
-        msg = "No NVIDIA GPUs available on the remote machine"
-        raise exp.GroqModelRuntimeError(msg)
+def setup_remote_host(client, device_type: str, output_dir: str) -> None:
+    if device_type == "gpu":
+        # Check if at least one NVIDIA GPU is available remotely
+        stdout, exit_code = exec_command(client, "lspci | grep -i nvidia")
+        if stdout == "" or exit_code == 1:
+            msg = "No NVIDIA GPUs available on the remote machine"
+            raise exp.GroqModelRuntimeError(msg)
+        files_to_transfer = ["execute-gpu.py"]
+    elif device_type == "cpu":
+        # Check if x86_64 CPU is available remotely
+        stdout, exit_code = exec_command(client, "uname -i")
+        if stdout != "x86_64" or exit_code == 1:
+            msg = "Only x86_64 CPUs are supported at this time for competitive benchmarking"
+            raise exp.GroqModelRuntimeError(msg)
+        files_to_transfer = ["execute-cpu.py", "setup_ort_env.sh"]
+    else:
+        raise ValueError(f"Only 'cpu' and 'gpu' are supported. But received {device_type}")
 
     # Transfer common files to host
-    exec_command(client, "mkdir mlagility_remote_cache", ignore_error=True)
+    exec_command(client, f"mkdir {output_dir}", ignore_error=True)
     dir_path = os.path.dirname(os.path.realpath(__file__))
     with MySFTPClient.from_transport(client.get_transport()) as s:
-        s.put(f"{dir_path}/execute-gpu.py", "mlagility_remote_cache/execute-gpu.py")
-
+        for file in files_to_transfer:
+            s.put(f"{dir_path}/{file}", f"{output_dir}/{file}")
+
+def setup_local_host(device_type: str, output_dir: str) -> None:
```

Comment:
Seems weird to me that a function like this is in a module named `cloud`. Should this module be renamed to `device`?